### PR TITLE
solve issue 64,87

### PR DESCRIPTION
**Pull request by Seeyn**
Add function to resize the original image to solve issue 64.
Modify load function to autofill word cloud to solve issue 87.

### CODE REVIEW

File: kumo-core/src/main/java/com/kennycason/kumo/ImageProcessor.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * @param width is the resized image's width
+     * @param height is the resized image's height
+     * @return the resized image in ByteArrayInputStream form
+     */
+    public static InputStream readImage(String fileName, int width, int height, String imageType) throws IOException, InterruptedException {
+        BufferedImage originImage;
+        originImage = ImageIO.read(getInputStream(fileName));
+
+        Image scaledImage = originImage.getScaledInstance(width, height, Image.SCALE_SMOOTH);
+        BufferedImage bufferedImage;
```
Comment (kennycason) [inline kumo-core/src/main/java/com/kennycason/kumo/ImageProcessor.java:36]: same here

File: kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java [hunk 2]
```diff
@@ -81,11 +81,55 @@ public FrequencyAnalyzer() {
...
+        final List<WordFrequency> wordFrequencies = new ArrayList<>();
+
+        final Map<String, Integer> cloud = buildWordFrequencies(texts, wordTokenizer);
+        int totalLength = 0;
+        for (Map.Entry<String, Integer> entry : cloud.entrySet()) {
+            totalLength += entry.getKey().length();
+        }
+        if (totalLength == 0){
+            cloud.put(autoFillWord,1);
+            totalLength = 7;
```
Comment (kennycason) [inline kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java:125]: Did you mean `totalLength = autoFillWorld.length();`？

File: kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java [hunk 3]
```diff
@@ -81,11 +81,55 @@ public FrequencyAnalyzer() {
...
+            for (int i = 0; i < Math.max(wordFrequenciesToReturn / totalLength,1); i++) {
```
Comment (kennycason) [inline kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java:126]:
```java
final int timesToAdd = Math.max(wordFrequenciesToReturn / totalLength, 1);
for (int i = 0; i < timesToAdd; i++) { ... 
```
I recommend pulling this condition out of the for-loop to help performance a bit.

File: kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java [hunk 4]
```diff
@@ -81,11 +81,55 @@ public FrequencyAnalyzer() {
...
+        return load(IOUtils.readLines(inputStream, characterEncoding), autoFill, autoFillWord);
+    }
+    public List<WordFrequency> load(final InputStream inputStream, boolean autoFill) throws IOException {
+        return load(IOUtils.readLines(inputStream, characterEncoding), autoFill, "nothing");
+    }
+    public List<WordFrequency> load(final List<String> texts, boolean autoFill){
+        return load(texts, autoFill, "nothing");
+    }
+
+    public List<WordFrequency> load(final List<String> texts, boolean autoFill, String autoFillWord) {
```
Comment (kennycason) [inline kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java:111]: 我觉得这里有个COMMENT，解释一下你算法会帮助点。谢谢！

File: kumo-core/src/test/java/com/kennycason/kumo/IssueTest.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+import java.awt.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.List;
+
+public class IssueTest {
+
+    //我们要有四个测试样例，分别对应没有文字，自动填充，文字过少，复制扩充
+    //图像大小不合适，重新调整，小图调大，和大图调小，都要测试一次
```
Comment (kennycason) [inline kumo-core/src/test/java/com/kennycason/kumo/IssueTest.java:24]: 我看到这种中文的COMMENT我一直觉得中文表达能力很高。句子又短又清楚，效果很好，对于个语言来说真的太棒。虽然我母语是英文我有时候希望中文会变世界上第一个语言。

**Comment by Seeyn** [comment]
We are glad to recieve your reply .We have followed your suggestions .And if you have any other suggestions, please tell us. :)

File: kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java [hunk 6]
```diff
@@ -104,12 +104,18 @@ public FrequencyAnalyzer() {
...
     }
-
+    //If autoFill is false, just call load
+    //Else if there are some words to draw, repeat them until the number of characters is similar with wordFrequenciesToReturn
+    //Else repeat autoFillWord to satisfy the drawing cloud need.
+    //Note: the repeat time is dependence on the sum of word length. We consider character number is more important.
+    //Which means long word will be repeated less
     public List<WordFrequency> load(final List<String> texts, boolean autoFill, String autoFillWord) {
-        if(!autoFill){
+        if(autoFill == false){
```
Comment (kennycason) [inline kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java:118]: it's ok for now, but for future `if (!autoFill) {` was 100% ok with me. :)

**Comment by kennycason** [comment]
@Seeyn hi! thanks for the fast updates! I'll merge and deploy this over the weekend. :)

File: kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java (hunk 6 above)
Comment (Seeyn) [reply kumo-core/src/main/java/com/kennycason/kumo/nlp/FrequencyAnalyzer.java]: got it

**Comment by Seeyn** [comment]
@kennycason 好的,祝你有个愉快的周末

**Comment by kennycason** [comment]
@Seeyn 谢谢🙏！

**Comment by kennycason** [comment]
Merged into master. Version released. Current version is 1.27, it should show up in maven central shortly. Thanks!

**Pull request merged**